project/src/components/cities-list/cities-list.tsx

```diff
@@ -0,0 +1,32 @@
+import { useAppDispatch } from '../../hooks';
+import { setCityName } from '../../store/reducers/city-reducer';
+import {Settings} from '../../utils/const';
+
+function CityList(props: { city: string }) {
+  const {city} = props;
+  const {CITYES} = Settings;
+  const dispatch = useAppDispatch();
+
+  function handleClick(cityName: string) {
+    return () => dispatch(setCityName(cityName));
+  }
+
+  return (
+    <section className="locations container">
+      <ul className="locations__list tabs__list">
+        {CITYES.map((cityName) => {
+          const className = `locations__item-link tabs__item${cityName === city && ' tabs__item--active'}`;
+          return (
+            <li key={cityName} className="locations__item" onClick={handleClick(cityName)}>
```

Comment:
Обычно так не пишут, потому что в таком случае неочевидно что в результате вызова этой функции у тебя здесь будет колбек. И само название не говорит об этом лучше перенеси тело функции прямо сюда
```suggestion
            <li key={cityName} className="locations__item" onClick={() => dispatch(setCityName(cityName))}>
```
В целом вынос хендлеров в отдельную функцию это хороший тон, например для использования с хуком useCallback но это немного другая история

Reply:
исправил